examples/run/sycl/full_chain_algorithm.sycl

```diff
@@ -35,8 +35,32 @@ namespace traccc::sycl {
 namespace details {
 
 struct full_chain_algorithm_data {
+
+    /// Constructor
+    explicit full_chain_algorithm_data(const ::sycl::async_handler& handler)
+        : m_queue{handler} {}
+
+    /// Destructor
+    ~full_chain_algorithm_data() {
+        // Delete the objects on the heap explicitly, so that they would
+        // definitely be deleted before the SYCL queue object. Even take care
+        // of deleting them in the correct order.
+        m_cached_device_mr.reset();
+        m_device_mr.reset();
+    }
```

Comment:
This destructor is strictly unnecessary, as members are automatically destroyed in reverse declaration order.